project/Omnidoc.scala

```diff
@@ -0,0 +1,57 @@
+import sbt.Keys._
+import sbt.Package.ManifestAttributes
+import sbt._
+
+/**
+ * Copied in from
+ * https://github.com/playframework/interplay/blob/master/src/main/scala/interplay/Omnidoc.scala
+ *
+ * This AutoPlugin adds the `Omnidoc-Source-URL` key on the MANIFEST.MF of artifact-sources.jar so
+ * later Omnidoc can use that value to link scaladocs to GitHub sources.
+ */
+object Omnidoc extends AutoPlugin {
+
+  object autoImport {
+//    lazy val omnidocGithubRepo = settingKey[String]("Github repository for source URL")
+    lazy val omnidocSnapshotBranch = settingKey[String]("Git branch for development versions")
+//    lazy val omnidocTagPrefix = settingKey[String]("Prefix before git tagged versions")
+    lazy val omnidocPathPrefix = settingKey[String]("Prefix before source directory paths")
+    lazy val omnidocSourceUrl  = settingKey[Option[String]]("Source URL for scaladoc linking")
+  }
+
+  val omnidocGithubRepo = Some(s"playframework/${Common.repoName}")
+  val omnidocTagPrefix  = Some("")
+
+  val SourceUrlKey = "Omnidoc-Source-URL"
+
+  override def requires = sbt.plugins.JvmPlugin
+
+  override def trigger = noTrigger
+
+  import autoImport._
+
+  override def projectSettings = Seq(
+    omnidocSourceUrl := omnidocGithubRepo.map { repo =>
+      val development: String = (omnidocSnapshotBranch ?? "master").value
+      val tagged: String      = omnidocTagPrefix.getOrElse("v") + version.value
+      val tree: String        = if (isSnapshot.value) development else tagged
+      val prefix: String      = "/" + (omnidocPathPrefix ?? "").value
+      val path: String = {
+        val buildDir: File      = (baseDirectory in ThisBuild).value
+        val projDir: File       = baseDirectory.value
+        val rel: Option[String] = IO.relativize(buildDir, projDir)
+        rel match {
+          case None if buildDir == projDir => ""                // Same dir (sbt 0.13)
+          case Some("")                    => ""                // Same dir (sbt 1.0)
```

Comment:
Probably we can get rid of these two cases in `play-json`

Reply:
I wanted to keep the diff to Interplay's version minimal.